### PR TITLE
Change waiting time for a pod in e2e test

### LINKED ISSUE #36586: [k8s.io] Kubectl client [k8s.io] Kubectl should be able to talk to api server kubectl running in a pod could talk to api server [Conformance] {Kubernetes e2e suite}
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-e2e-gci-gke/3658/

Failed: [k8s.io] Kubectl client [k8s.io] Kubectl should be able to talk to api server kubectl running in a pod could talk to api server [Conformance] {Kubernetes e2e suite}

```
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:585
Expected
    <*errors.errorString | 0xc4203ab040>: {
        s: "timed out waiting for the condition",
    }
to be nil
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:579
```

### PR DESCRIPTION
**Pull request by mengqiy**
Change waiting time for a pod.

partially addresses: #36962

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36777)

### CODE REVIEW

File: test/e2e/kubectl.go [hunk 1]
```diff
@@ -568,7 +568,7 @@ var _ = framework.KubeDescribe("Kubectl client", func() {
 			podJson := readTestFileOrDie(kubectlInPodFilename)
 			By("validating api verions")
 			framework.RunKubectlOrDieInput(string(podJson), "create", "-f", "-", nsFlag)
-			err := wait.PollImmediate(time.Second, time.Minute, func() (bool, error) {
+			err := wait.PollImmediate(3*time.Second, 3*time.Minute, func() (bool, error) {
```
Comment (ixdy) [inline test/e2e/kubectl.go]: should it really be taking more than a minute for this pod to start?

**Comment by ymqytw** [comment]
Updated the code. Added debugging code.

**Comment by ymqytw** [comment]
@ixdy ping

**Comment by ymqytw** [comment]
Reopen this PR. Since we still need to give pod more time as other tests do.

cc: @dims

**Comment by dims** [comment]
LGTM 👍  @ymqytw (pending tests, but hey 5 mins is better than one!)

**Comment by ymqytw** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by ixdy** [comment]
#36586 is 1.5, so I'm guessing this should be too?

**Comment by ixdy** [comment]
I'm happy that it looks like we found the actual bug, too. :)

**Comment by ymqytw** [comment]
@k8s-merge-robot Rebased.

**Comment by ymqytw** [comment]
Closing because this test is being reverted.

**Comment by saad-ali** [comment]
This was a change to the common framework, does it not apply to other tests using this method?

**Pull request closed without merging**